twenty-first/src/math/polynomial.rs

```diff
@@ -765,11 +885,378 @@ where
     /// # Panics
     ///
     /// Panics if the `divisor` is zero.
-    pub fn divide(&self, divisor: &Self) -> Self {
-        // According to `cargo bench --bench poly_div` on mjolnir, naive division is always faster
-        // than fast division.
-        let (quotient, _) = self.naive_divide(divisor);
-        quotient
+    pub fn divide(&self, divisor: &Self) -> (Self, Self) {
+        const FAST_DIVIDE_THRESHOLD: isize = 1000;
+        if self.degree() > FAST_DIVIDE_THRESHOLD {
+            self.fast_divide(divisor)
+        } else {
+            self.naive_divide(divisor)
+        }
+    }
+
+    /// Compute a polynomial g(X) from a given polynomial f(X) such that
+    /// g(X) * f(X) = 1 mod X^n , where n is the precision.
+    ///
+    /// In formal terms, g(X) is the approximate multiplicative inverse in
+    /// the formal power series ring, where elements obey the same
+    /// algebraic rules as polynomials do but can have an infinite number of
+    /// coefficients. To represent these elements on a computer, one has to
+    /// truncate the coefficient vectors somewhere. The resulting truncation
+    /// error is considered "small" when it lives on large powers of X. This
+    /// function works by applying Newton's method in this ring.
+    ///
```

Comment:
```suggestion
    ///
    /// # Example
    ///
    /// ```
    /// # use num_traits::One;
    /// # use twenty_first::prelude::*;
    /// let precision = 8;
    /// let f = Polynomial::new(bfe_vec![42; precision]);
    /// let g = f.formal_power_series_inverse_newton(precision);
    /// let x_to_the_n = Polynomial::one().shift_coefficients(precision);
    /// let (_quotient, remainder) = g.multiply(&f).divide(&x_to_the_n);
    /// assert!(remainder.is_one());
    /// ```
```